Referencables verifer

sirherrbatka opened the pull request:
There is no tabs here. If there is odd formatting, could you give me the concrete line number?

sirherrbatka committed:
Merge pull request #6 from zinnschlag/master

updating fork

sirherrbatka committed:
Added Referencable checks class, added method to get refidcontainer, added method to get mBooks.

Currently only books are checked, and only if name is present.

sirherrbatka committed:
Added aditional checks for books. Activator check.

sirherrbatka committed:
added getpotions

sirherrbatka committed:
Renamed getActivator to getActivators.

sirherrbatka committed:
Small optimization.

sirherrbatka committed:
Getting size of potions.

sirherrbatka committed:
Adde potionCheck method

sirherrbatka committed:
Added enchantmnet check for books.

sirherrbatka committed:
Works, but it seems that I miss something in referencables. I will write post on the forum.

sirherrbatka committed:
Added check for non-positive quality of alchemical apparatus

sirherrbatka committed:
Added armor check

sirherrbatka committed:
Added armor check and related stuff.

sirherrbatka committed:
added getContainers

sirherrbatka committed:
added mContainersSize.

sirherrbatka committed:
added container check

sirherrbatka committed:
Added getCreatures

sirherrbatka committed:
added mCreaturesSize

sirherrbatka committed:
Added creatureCheck. I don't know meaning of all data fields here.

sirherrbatka committed:
Added door check with related methods.

sirherrbatka committed:
Added ingredient check

sirherrbatka committed:
Working on leveled lists now.

sirherrbatka committed:
added creature leveled list check

sirherrbatka committed:
checkin leveled items lists

sirherrbatka committed:
removed set sizes, consted sizes, added itemleveledlist check

sirherrbatka committed:
NPC. Still WIP.

sirherrbatka committed:
Code compiles now

sirherrbatka committed:
still working on npc check

sirherrbatka committed:
Corrections

sirherrbatka committed:
Small fix for lights.

sirherrbatka committed:
small oops + added new variables to the constructor

sirherrbatka committed:
Last fixes.

sirherrbatka committed:
removing member size values, since this does not work properly.

sirherrbatka committed:
Added faction check.

sirherrbatka committed:
Dynamic_casting, checking rank. Commented out faction check, since it will not work.

sirherrbatka committed:
added comment

sirherrbatka committed:
Check for head and hair. Correct faction check.

sirherrbatka committed:
reformatted

zinnschlag commented:
I don't see any tab problems in this version. However what I see are random changes in the source formatting (inserting/deleting space and such) of otherwise unchanged code, which make it pretty much impossible to review your code (have a look at the diff; a total mess). Not to mention that these formatting changes are not covered by our guidelines, which means the next person who works on the code may just change them back.

It is okay to make beautifying adjustments to the lines you are working on anyway, but what I see here is a no go.

I did manage to find one more actual problem with the code: You are manually going through the collections to check if an ID exists. We have the searchId function for that, which will also take care of case issues.

Also, the code is very repetitive, which is definitely not a good thing (some templates might come in handy).

sirherrbatka committed:
correcting

sirherrbatka committed:
reverting refiddata.cpp

sirherrbatka committed:
reverting refidata.hpp

sirherrbatka commented:
Corrected formatting, replaced manual loop with searchId.

I guess that this problem is a potential use case for templates, however ─ I 
didn't learned templates (yet). It will take quite a lot of time for me to 
finish with learning git-fu, dealing with some personal issues and then move to 
templates. If you don't mind waiting for those, just close pull request and be 
patient ;-)

sirherrbatka committed:
Corrected stupid typo.

zinnschlag commented:
Sure. No hurry. But we can leave the pull request open. Its not like it takes up space or anything.

sirherrbatka commented:
Ok, It's nice to have a real, production code to train new skill, and code 
review at the same time ;-)

sirherrbatka committed:
Inventory item template.

sirherrbatka committed:
Finishing stuff.

sirherrbatka committed:
style corrections

sirherrbatka committed:
Some changes in weapons check.

lgromanowski commented on apps/opencs/model/tools/referenceablecheck.cpp:
```diff
@@ -0,0 +1,984 @@
+    //Don't know what unknown is for
+    int Gold(NPC.mNpdt52.mGold);
+
+    if (NPC.mNpdtType == 12) //12 = autocalculated
```
Use enum instead of plain "12"

lgromanowski commented on apps/opencs/model/tools/referenceablecheck.hpp:
```diff
@@ -0,0 +1,54 @@
+    class ReferenceableCheckStage : public CSMDoc::Stage
+    {
+        public:
+            ReferenceableCheckStage(const CSMWorld::RefIdData& referenceable, const CSMWorld::IdCollection<ESM::Race>& races, const CSMWorld::IdCollection<ESM::Class>& classes, const CSMWorld::IdCollection<ESM::Faction>& factions);
```
Please break long lines near 80-100 column.

zinnschlag replied:
More 100 than 80. 80 as a character per line limit is rather archaic.

sirherrbatka committed:
Splited some long argument lists to the multiple lines.

sirherrbatka committed:
Various style corrections.

sirherrbatka committed:
implemented list of magical bolts for skipping.

sirherrbatka committed:
replaced raw values with enums.

sirherrbatka committed:
changed ID check in leveled list to name.

sirherrbatka committed:
Removing name check from listCheck. Id can't be empty to be sure.

sirherrbatka committed:
Added proper id check.

sirherrbatka committed:
Chainging names to comply our policy.

sirherrbatka committed:
forgot to save the file before comit -_-'

sirherrbatka committed:
npc instead of NPC.

sirherrbatka committed:
static is a keyword. renamed static to staticElement

sirherrbatka committed:
corrected one, additional name to follow policy

sirherrbatka committed:
added final check for player npc. Removed useless includes. However, this code spams exceptions and I can't figure out why.

sirherrbatka committed:
Changes according to the comment.

sirherrbatka committed:
Shameful bug fixed.

sirherrbatka committed:
splited long lines in the header.

sirherrbatka committed:
camel case in the header

sirherrbatka committed:
Splited another long line in the header.

sirherrbatka committed:
replaced == operator for string comparsion in npc check with boost algorithm to get case insensitive check.

sirherrbatka committed:
Final check is not performed with just +1! Something is not right.

sirherrbatka committed:
fixed bug resposnsible for exception throwed.

sirherrbatka committed:
Creature check was not invoked.

sirherrbatka committed:
changed according to the scrawl sugestion

zinnschlag commented on apps/opencs/model/tools/referenceablecheck.cpp:
```diff
@@ -0,0 +1,1107 @@
+#include "../world/record.hpp"
+#include "../world/universalid.hpp"
+#include <boost/algorithm/string.hpp>
+#include "../../../../openmw/apps/openmw/mwclass/misc.hpp"
```
This line is wrong. First it doesn't compile. Second, you are not supposed to cross-include between different applications in the apps directory.

sirherrbatka replied:
I blame scrawl. ;-)

sirherrbatka committed:
Moving back to boost for case insensitive comparsion.

sirherrbatka committed:
Removed boost in favor of #include <components/misc/stringops.hpp>. Hopefully this will make scrawl happy. :P

zinnschlag commented:
Not yet there.
1. I am getting tons of errors about NPCs having an invalid race. Looks like a bug.
2. I think lights with 0 duration are fine (burned out light sources?).

sirherrbatka commented:
@1 Can't replicate this on my branch.
@2 Burned out lights makes no sense in the morrowind since after getting 
burned out they are removed from the inventory. Using 0 duration light would 
obviously instantly remove it from the inventory ─ very surprising and 
unexpected behavior. Therefore lights should have either negative, or positive 
duration.

ghost commented:
Hm, uesp says:
"A value of 0 indicates lights which are already burned out."
And lists light_com_torch_burnedout_01 with a time of 0.
Perhaps our implementation is wrong? Should burned out torches maybe not be removed from the inventory? Did anyone test that?

sirherrbatka commented:
This is vanilla behavior. There are only 3 burned out lights in the 
morrowind.esm and I'm not even sure if they are actually placed in game 
itself.

ghost commented:
Apparently if you add light_com_torch_burnedout_01 to your inventory and equip it, vanilla just crashes.
Guess you are right, then.

sirherrbatka commented:
Yeah, Morrowind.esm has a bit of broken stuff inside of it.

Anyway, @zinnschlag do you have those errors after running verifer on morrowind.esm or any other esm file?

zinnschlag commented:
Morrowind.esm. My copy of Morrowind.esm. Different versions/localisations of that file causing problems again, maybe?

sirherrbatka committed:
Localised version of morrowind will no longer spam false positives. Don't check diff, please.

sirherrbatka commented:
Dnia wtorek, 14 stycznia 2014 06:37:25 Marc Zinnschlag pisze:

> Morrowind.esm. My copy of Morrowind.esm. Different versions/localisations of
> that file causing problems again, maybe?
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/zinnschlag/openmw/pull/1195#issuecomment-32269137

Yes, I see. Extreamly stupid mistake on my side. Solved now.

sirherrbatka committed:
ooops, using search id

sirherrbatka committed:
Corrected brackets.

sirherrbatka committed:
Being any idiot is hard.
